### PR TITLE
Use PartialFunction for more concise code.

### PR DESCRIPTION
**Pull request by nickchapman-da**
Use scala `PartialFunction` for more concise code in `ValidationSpec.scala`

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/validation/ValidationSpec.scala [hunk 1]
```diff
@@ -354,57 +338,39 @@ class ValidationSpec extends AnyFreeSpec {
...
+    tweakOptLocation.run(nf.optLocation).map { x => nf.copy(optLocation = x) }
   }
-  private val tweakFetchActingPartiesEmpty = Tweak[Node] { //insig
-    case nf: Node.NodeFetch[_] =>
-      if (nf.actingParties.isEmpty) {
-        tweakPartySet.run(nf.actingParties).map { x => nf.copy(actingParties = x) }
-      } else Nil
-    case _ => Nil
+  private val tweakFetchActingPartiesEmpty = Tweak[Node] {
+    case nf: Node.NodeFetch[_] if (nf.actingParties.isEmpty) => //insig
```
Comment (remyhaemmerle-da) [inline daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/validation/ValidationSpec.scala:351]:
```suggestion
    case nf: Node.NodeFetch[_] if nf.actingParties.isEmpty => //insig
```
parenthesis are not required in guard

**Review by cocreature — APPROVED** [review]
thx!

File: daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/validation/ValidationSpec.scala [hunk 2]
```diff
@@ -524,30 +467,24 @@ class ValidationSpec extends AnyFreeSpec {
             ne.copy(exerciseResult = None),
           )
       }
-    case _ => Nil
```
Comment (remyhaemmerle-da) [inline daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/validation/ValidationSpec.scala:527]:
```
  private val tweakExerciseExerciseResultSome = Tweak[Node] {
    case ne: Node.NodeExercises[_, _] if ne.exerciseResult.isDefined => //sig
          List(
            ne.copy(exerciseResult = ne.exerciseResult.map(changeValue)),
            ne.copy(exerciseResult = None),
          )
      }
```

**Review by remyhaemmerle-da — APPROVED** [review]
:+1:

File: daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/validation/ValidationSpec.scala (hunk 1 above)
Comment (nickchapman-da) [reply daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/validation/ValidationSpec.scala]: done

File: daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/validation/ValidationSpec.scala (hunk 2 above)
Comment (nickchapman-da) [reply daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/validation/ValidationSpec.scala:527]: nope. one step too far. ;)

**Pull request merged**